checks.d/mapreduce.py

```diff
@@ -0,0 +1,500 @@
+'''
+MapReduce Job Metrics
+---------------------
+mapreduce.job.elapsed_ime                The elapsed time since the application started (in ms)
+mapreduce.job.maps_total                 The total number of maps
+mapreduce.job.maps_completed             The number of completed maps
+mapreduce.job.reduces_total              The total number of reduces
+mapreduce.job.reduces_completed          The number of completed reduces
+mapreduce.job.maps_pending               The number of maps still to be run
+mapreduce.job.maps_running               The number of running maps
+mapreduce.job.reduces_pending            The number of reduces still to be run
+mapreduce.job.reduces_running            The number of running reduces
+mapreduce.job.new_reduce_attempts        The number of new reduce attempts
+mapreduce.job.running_reduce_attempts    The number of running reduce attempts
+mapreduce.job.failed_reduce_attempts     The number of failed reduce attempts
+mapreduce.job.killed_reduce_attempts     The number of killed reduce attempts
+mapreduce.job.successful_reduce_attempts The number of successful reduce attempts
+mapreduce.job.new_map_attempts           The number of new map attempts
+mapreduce.job.running_map_attempts       The number of running map attempts
+mapreduce.job.failed_map_attempts        The number of failed map attempts
+mapreduce.job.killed_map_attempts        The number of killed map attempts
+mapreduce.job.successful_map_attempts    The number of successful map attempts
+
+MapReduce Job Counter Metrics
+-----------------------------
+mapreduce.job.counter.reduce_counter_value   The counter value of reduce tasks
+mapreduce.job.counter.map_counter_value      The counter value of map tasks
+mapreduce.job.counter.total_counter_value    The counter value of all tasks
+
+MapReduce Map Task Metrics
+--------------------------
+mapreduce.job.map.task.progress     The distribution of all map task progresses
+mapreduce.job.map.task.elapsed_time The distribution of all map tasks elapsed time
+
+MapReduce Reduce Task Metrics
+--------------------------
+mapreduce.job.reduce.task.progress      The distribution of all reduce task progresses
+mapreduce.job.reduce.task.elapsed_time  The distribution of all reduce tasks elapsed time
+'''
+
+# stdlib
+from urlparse import urljoin
+
+# 3rd party
+import requests
+from requests.exceptions import Timeout, HTTPError, InvalidURL, ConnectionError
+from simplejson import JSONDecodeError
+
+# Project
+from checks import AgentCheck
+
+# Service Check Names
+SERVICE_CHECK = 'mapreduce.cluster.can_connect'
+YARN_SERVICE_CHECK = 'mapreduce.resource_manager.can_connect'
+MAPREDUCE_SERVICE_CHECK = 'mapreduce.application_master.can_connect'
```

Comment:
Actually hadn't noticed that there were different service checks when I suggested to only send an `OK` once.

The check should actually send an `OK` service check for each service kind when all the requests to that service have been made successfully.

So from what I see in the current state of the check, we should probably:
- remove the `SERVICE_CHECK` service check altogether (given that we only ever send `OK`s on it)
- send an `OK` `YARN_SERVICE_CHECK` service check right after `self._get_running_app_ids`, because, at this point of the check, all the requests made to the resource manager have been successful
- send an `OK` `MAPREDUCE_SERVICE_CHECK` at the end of the main `check` method, because it's only at that point of the check that we know that all the requests to the application master have been successful.

Does that sound sensible to you? If so could you make these changes and add a small comment on each `OK` service check explaining why we're sending the service check at that point of the check?

Thanks!